examples/exampleutil/exampleutil.go

```diff
@@ -1,28 +1,93 @@
 package exampleutil
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"log"
+	"math/rand"
+	"net/http"
 	"strings"
 	"time"
 
-	"math/rand"
-
 	"github.com/bitcoin-sv/spv-wallet-go-client/config"
 )
 
-var ExampleConfig = config.NewDefaultConfig("http://localhost:3003")
+// NewDefaultConfig returns a new instance of the default example configuration.
+func NewDefaultConfig() config.Config {
+	return config.NewConfig()
+}
+
+// NewCustomConfig allows creating a custom configuration with optional parameters.
+func NewCustomConfig(addr string, timeout time.Duration) config.Config {
+	options := []config.Option{}
+
+	if addr != "" {
+		options = append(options, config.WithAddr(addr))
+	}
+
+	if timeout > 0 {
+		options = append(options, config.WithTimeout(timeout))
+	}
+
+	return config.NewConfig(options...)
+}
+
+// NewCustomConfig allows creating a custom configuration with optional parameters.
+func NewCustomConfigWithTransport(addr string, timeout time.Duration, transport *http.Transport) config.Config {
+	options := []config.Option{}
+
+	if addr != "" {
+		options = append(options, config.WithAddr(addr))
+	}
+
+	if timeout > 0 {
+		options = append(options, config.WithTimeout(timeout))
+	}
+
+	if transport != nil {
+		options = append(options, config.WithTransport(transport))
+	}
+
+	return config.NewConfig(options...)
```

Comment:
Once zero value checks are applied within each functional option, these extra checks will no longer be necessary. This should help streamline the code.

What’s the purpose of using these wrapper functions instead of defining specific configuration structures?  Perhaps creating an example to demonstrate how these options could be used would make this approach easier to understand. What do you think? 🤔